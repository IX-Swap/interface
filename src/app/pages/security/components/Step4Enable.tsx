import React from 'react'
import { Typography, Grid } from '@mui/material'
import { Enable2faFormValues } from 'app/pages/security/types'
import { Form } from 'components/form/Form'
import { StepWrapper } from 'app/pages/security/components/StepWrapper'
import { EnableFormFields } from 'app/pages/security/components/EnableFormFields'
import { useEnable2fa } from 'app/pages/security/hooks/useEnable2fa'

const defaultValues = {
  otp: ''
}

export interface Step4EnableProps {
  nextStep: () => void
}

export const Step4Enable = ({ nextStep }: Step4EnableProps) => {
  const [enable2fa] = useEnable2fa(nextStep)
  const onSubmit = async (values: Enable2faFormValues) => {
    await enable2fa(values)
  }

  return (
    <StepWrapper title='Enable Authenticator App'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please enter the 6 digit code from your authenticator app
          </Typography>
        </Grid>
        <Grid item>
          <Form
            defaultValues={defaultValues}
            onSubmit={onSubmit}
            data-testid='form'
          >
            <EnableFormFields />
          </Form>
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Once enabled your authenticator you will be logged out and re-enter
            the OTP in order to access the platform
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  )
}
