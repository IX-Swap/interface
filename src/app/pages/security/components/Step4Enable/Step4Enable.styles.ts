import { makeStyles } from '@mui/styles'

export const useStyles = makeStyles(theme => ({
  container: {},
  text: {
    fontSize: 16,
    opacity: 0.8,
    margin: theme.spacing(2, 0, 23),

    [theme.breakpoints.down('md')]: {
      margin: theme.spacing(3, 0, 5)
    }
  }
}))
