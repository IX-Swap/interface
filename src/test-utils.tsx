import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
