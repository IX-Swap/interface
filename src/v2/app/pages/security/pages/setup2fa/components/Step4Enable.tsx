//
import React from 'react'
import { Container, Typography, Box, Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

import { createTypedForm } from 'v2/components/form/createTypedForm'
import { useEnable2fa } from '../hooks/useEnable2fa'
import { Enable2faFormValues } from '../types'
import { enable2faFormValuesSchema } from '../validation'

const useEnable2faForm = createTypedForm<Enable2faFormValues>()

const useStyles = makeStyles(() => ({
  textField: {
    width: '100%'
  }
}))

const defaultValues = {
  otp: ''
}

export const Step4Enable = () => {
  const classes = useStyles()
  const { Form, TextField, Submit } = useEnable2faForm()
  const [enable2fa] = useEnable2fa()
  const onSubmit = async (values: Enable2faFormValues) => {
    await enable2fa(values)
  }

  return (
    <Container>
      <Typography align='center'>Enable your Google Authenticator</Typography>
      <Grid container justify='center'>
        <Box mt={4} width='30%'>
          <Form
            defaultValues={defaultValues}
            validationSchema={enable2faFormValuesSchema}
            onSubmit={onSubmit}
            data-testid='form'
          >
            <TextField
              label='OTP'
              name='otp'
              inputProps={{
                autoComplete: 'off',
                className: classes.textField
              }}
            />
            <Box my={4} width='100%' textAlign='center'>
              <Submit>Submit</Submit>
            </Box>
          </Form>
        </Box>
      </Grid>
    </Container>
  )
}
