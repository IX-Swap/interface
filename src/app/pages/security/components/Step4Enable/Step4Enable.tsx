import React from 'react'
import { Typography, Grid } from '@mui/material'
import { Enable2faFormValues } from 'app/pages/security/types'
import { Form } from 'components/form/Form'
import { StepWrapper } from 'app/pages/security/components/StepWrapper'
import { EnableFormFields } from 'app/pages/security/components/EnableFormFields/EnableFormFields'
import { useEnable2fa } from 'app/pages/security/hooks/useEnable2fa'
import { useStyles } from './Step4Enable.styles'

const defaultValues = {
  otp: ''
}

export interface Step4EnableProps {
  nextStep: () => void
  update2FA?: boolean
}

export const Step4Enable = ({
  nextStep,
  update2FA = false
}: Step4EnableProps) => {
  const classes = useStyles()
  const [enable2fa, { isError }] = useEnable2fa(nextStep, !update2FA)
  const onSubmit = async (values: Enable2faFormValues) => {
    await enable2fa(values)
  }

  return (
    <StepWrapper
      title={
        update2FA ? 'Update Authenticator App' : 'Enable Authenticator App'
      }
    >
      <Grid container justifyContent={'center'}>
        <Grid
          container
          direction='column'
          alignItems='center'
          className={classes.container}
        >
          <Grid item>
            <Typography align='center' variant='body1' className={classes.text}>
              Please enter the 6 digit code from your Authenticator App
            </Typography>
          </Grid>
          <Grid item>
            <Form
              defaultValues={defaultValues}
              onSubmit={onSubmit}
              data-testid='form'
            >
              <EnableFormFields isError={isError} />
            </Form>
          </Grid>
        </Grid>
      </Grid>
    </StepWrapper>
  )
}
