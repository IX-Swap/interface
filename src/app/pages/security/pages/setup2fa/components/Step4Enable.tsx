import React from 'react'
import { Typography, Box, Grid } from '@mui/material'
import { useEnable2fa } from '../hooks/useEnable2fa'
import { Enable2faFormValues } from '../types'
import { enable2faFormValuesSchema } from '../validation'
import { Form } from 'components/form/Form'
import { Submit } from 'components/form/Submit'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'
import { EnableFormFields } from 'app/pages/security/pages/setup2fa/components/EnableFormFields'

const defaultValues = {
  otp: ''
}

export const Step4Enable = () => {
  const [enable2fa] = useEnable2fa()
  const onSubmit = async (values: Enable2faFormValues) => {
    await enable2fa(values)
  }

  return (
    <StepWrapper title='Enable Authenticator App'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please enter the 6 digit code from your authenticator app
          </Typography>
        </Grid>
        <Grid item>
          <Form
            defaultValues={defaultValues}
            validationSchema={enable2faFormValuesSchema}
            onSubmit={onSubmit}
            data-testid='form'
          >
            <EnableFormFields />
            <Box my={4} width='100%' textAlign='center'>
              <Submit>Enable</Submit>
            </Box>
          </Form>
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Once enabled your authenticator you will be logged out and re-enter
            the OTP in order to access the platform
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  )
}
