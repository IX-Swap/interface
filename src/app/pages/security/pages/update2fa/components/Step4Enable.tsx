import React from 'react'
import { Typography, Grid } from '@material-ui/core'
import { Enable2faFormValues } from '../types'
import { Form } from 'components/form/Form'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'
import { EnableFormFields } from 'app/pages/security/pages/update2fa/components/EnableFormFields'
import { useEnable2fa } from 'app/pages/security/pages/update2fa/hooks/useEnable2fa'

const defaultValues = {
  otp: ''
}

export interface Step4EnableProps {
  nextStep: () => void
}

export const Step4Enable = ({ nextStep }: Step4EnableProps) => {
  const [enable2fa] = useEnable2fa(nextStep)
  const onSubmit = async (values: Enable2faFormValues) => {
    await enable2fa(values)
  }

  return (
    <StepWrapper title='Enable Authenticator App'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please enter the 6 digit code from your authenticator app
          </Typography>
        </Grid>
        <Grid item>
          <Form
            defaultValues={defaultValues}
            onSubmit={onSubmit}
            data-testid='form'
          >
            <EnableFormFields />
          </Form>
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Once enabled your authenticator you will be logged out and re-enter
            the OTP in order to access the platform
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  )
}
