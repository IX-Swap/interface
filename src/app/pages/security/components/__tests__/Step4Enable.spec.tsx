import React from 'react'
import { render } from 'test-utils'
import { Step4Enable } from 'app/pages/security/components/Step4Enable/Step4Enable'

describe('Step4Enable', () => {
  const nextStep = jest.fn()

  it('should match snapshot', () => {
    const { container } = render(<Step4Enable nextStep={nextStep} />)
    expect(container).toMatchSnapshot()
  })
})
