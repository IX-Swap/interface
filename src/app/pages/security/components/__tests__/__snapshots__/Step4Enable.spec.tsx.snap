// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step4Enable should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-2xfiep-MuiTypography-root"
      >
        Enable Authenticator App
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-container-1 css-t0zib5-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-text-2 css-1gd5lqg-MuiTypography-root"
            >
              Please enter the 6 digit code from your Authenticator App
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <form
              data-testid="form"
              style="width: 100%;"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <div
                  class="makeStyles-container-5"
                  style="display: flex;"
                >
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <input
                      aria-label="Please enter verification code. Digit 1"
                      autocomplete="off"
                      class="makeStyles-input-7 "
                      maxlength="1"
                      placeholder="_"
                      style="width: 1em; text-align: center;"
                      type="tel"
                      value=""
                    />
                  </div>
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <input
                      aria-label="Digit 2"
                      autocomplete="off"
                      class="makeStyles-input-7"
                      maxlength="1"
                      placeholder="_"
                      style="width: 1em; text-align: center;"
                      type="tel"
                      value=""
                    />
                  </div>
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <input
                      aria-label="Digit 3"
                      autocomplete="off"
                      class="makeStyles-input-7"
                      maxlength="1"
                      placeholder="_"
                      style="width: 1em; text-align: center;"
                      type="tel"
                      value=""
                    />
                  </div>
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <input
                      aria-label="Digit 4"
                      autocomplete="off"
                      class="makeStyles-input-7"
                      maxlength="1"
                      placeholder="_"
                      style="width: 1em; text-align: center;"
                      type="tel"
                      value=""
                    />
                  </div>
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <input
                      aria-label="Digit 5"
                      autocomplete="off"
                      class="makeStyles-input-7"
                      maxlength="1"
                      placeholder="_"
                      style="width: 1em; text-align: center;"
                      type="tel"
                      value=""
                    />
                  </div>
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <input
                      aria-label="Digit 6"
                      autocomplete="off"
                      class="makeStyles-input-7"
                      maxlength="1"
                      placeholder="_"
                      style="width: 1em; text-align: center;"
                      type="tel"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-fzwid6"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root Mui-disabled makeStyles-button-3 css-gbhykm-MuiButtonBase-root-MuiButton-root"
                  disabled=""
                  tabindex="-1"
                  type="submit"
                >
                  Enable and Continue
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
